File: Class17/class_test.php
<?php

class Pet
{
    public function speak()
    {
        echo "....\n";
    }

    public function speakTo(string $name)
    {
        $this->speak();
        echo $name;
    }
}

class  Dog extends Pet
{
    public function speak()
    {
        echo "Wang!\n";
    }
}

function main()
{
    $dog = new Dog();
    $dog->speak();
    $dog->speakTo("肥龙");
}

main();

File: src/Class17/class_test.go
package Class17

import (
	"fmt"
	"testing"
)

type Pet struct {
}

func (p *Pet) Speak() {
	fmt.Println("...")
}

func (p *Pet) SpeakTo(name string) {
	p.Speak()
	fmt.Println(name)
}

type Dog struct {
	Pet
}

func (d *Dog) Speak() {
	fmt.Println("Wang!")
}

func TestClass(t *testing.T) {
	d := new(Dog)
	d.Speak()
	d.SpeakTo("肥龙") // 这里会输出 ... 肥龙，并不会出现 Wang!肥龙
}
